### ISSUE TITLE
New Descriptions for Balloon Hotkeys and Help Dialog

### ISSUE THREAD

**Issue by terracoda** [phetsims/balloons-and-static-electricity#141]
@jesse here's the content for the Balloon Hotkeys and Help.

H1: Balloon Hotkeys and Key Commands
H2: Keys for Dragging and Rubbing
Press the following letter keys to drag the Balloon in any direction with small steps.
- Press W to drag up.
- Press A to drag left.
- Press S to drag down.
- Press D to drag right.

Add the Shift key to a letter key to drag with big steps. 
- Shift plus W makes a big step up.
- Shift plus A makes a big step left, and so on.

H2: Hotkeys for Jumping 
Press the following hotkey combinations to jump the Balloon to a location in the Play Area.
- JS jumps close to Sweater.
- JW jumps to Wall.
- JN jumps to near Wall.
- JC jumps to center of Play Area.

H2: Release Balloon 
- Press Spacebar to release.
- Pressing the Tab key also releases the Balloon as you go to the next item.

H2: Grab Balloon button 
- Press Spacebar or Enter on the Grab Balloon button to pick up or grab the Balloon, then start dragging by pressing any of the following keys W, A, S, or D.

Press Escape or Close button to exit this help dialog.

**Comment by terracoda** [comment]
@jessegreenberg, how do you this reads? I can post the full help soon, too, but we need a new button for that. Do happy to have any ready-made button for a "How to use this sim" dialog?

**Comment by jessegreenberg** [comment]
Looks good to me @terracoda, but for the bullet

> Shift plus A makes a big step left, and so on.

Is this shorthand for the remaining letter combinations or is 'and so on' the text for the bullet point?

**Comment by terracoda** [comment]
@jessegreenberg I will look at the release behaviour when the Balloon hotkeys dialog is opened more carefully. I had forgotten about the effect of the explicit Grab button. Now that we have the Grab button, I think we can keep it as-is. The user gets a natural prompt to grab again. Once we have all help documentation in place and a new description for the button, I can revisit.

**Comment by terracoda** [comment]
"and so on." is meant to to keep the help text shorter :-) 
Thanks for asking. I love the way programmers think! Always identifying ambiguities.

**Comment by jessegreenberg** [comment]
I forgot to mention in the comment, I don't know of a 'How to use this sim' dialog or associated button that has been used an another simulation before.

> "and so on." is meant to to keep the help text shorter :-) 

Ok, great!  Thanks :)

> Now that we have the Grab button, I think we can keep it as-is.

Ok, sounds good.  I will leave it as is for now.

**Comment by terracoda** [comment]
Regarding the general help button, I will think about this, but can you direct me to where other PhET buttons live? PhET must have some visual design guidelines.

**Comment by jessegreenberg** [comment]
Hmm, I am not sure where a collection like this would be.  @arouinfar do you know of a comprehensive list, or list for PhET design guidelines for this kind of thing?

Ah, there is this list that we started: https://docs.google.com/document/d/12taDyvWwM9OcsUZsZEJaZEm7fukMaY40f-HTAK9Z39M/edit

Here is an image with a few example PhET buttons 
![image](https://cloud.githubusercontent.com/assets/6396244/14478632/ab874b7e-00d4-11e6-9f5d-89ef2efb626d.png)

**Comment by terracoda** [comment]
@jessegreenberg THANKS! This is great!

**Comment by terracoda** [comment]
Are there any design specification for the footer? The help link or button will be a link that is common across sims, kind of like the reset button and phet menu link.

**Comment by arouinfar** [comment]
@jessegreenberg @terracoda I can't think of a particular button style that would correspond to a help menu. This isn't something we've had to do before, so I think it would be safe to pick any style within the examples from @jessegreenberg's screenshot. I'll point out that the colors of any of those buttons can be changed.

**Comment by terracoda** [comment]
@arouinfar @jessegreenberg Thanks, I'll see what I can do with these :-)

**Comment by jessegreenberg** [comment]
Now that there is a visual representation, the whole keyboard help dialog should be written differently to make changing content and maintenance easier.  I will work on that next and include the new content in that process.

**Comment by terracoda** [comment]
@jessegreenberg, in Instance 16, the help dialog content does not match the content in #141. Did we agree on a new version? Or perhaps, it still needs to be updated?

Please note that I changed in the Hotkeys for Jumping. I changed JM to JC. One participant noted that the JM keys were difficult to operate and also JN and JM sound similar.

**Comment by jessegreenberg** [comment]
Yes, version 16 is likely out of date.  The most recent accessible version is 
http://www.colorado.edu/physics/phet/dev/html/balloons-and-static-electricity/1.2.0-reader.1/balloons-and-static-electricity_en.html

You can 'turn on' PhET's screen reader with the ?reader query parameter.
http://www.colorado.edu/physics/phet/dev/html/balloons-and-static-electricity/1.2.0-reader.1/balloons-and-static-electricity_en.html?reader

Thank you for pointing out the change to the jump hotkeys!  This is in the help dialog, but I have not yet implemented this change for the actual interaction.  I will create a new issue for that.

**Comment by jessegreenberg** [comment]
@terracoda mentioned that escape no longer closes the keyboard help dialog:

> @jessegreenberg Both the ESC key and the Close dialog button should be available to close the dialog. The ESC key is currently not working for me on Mac with Safari.

It probably isn't working for any platform at this point.  Should be fixed here.

**Comment by terracoda** [comment]
You are @jessegreenberg, the ESC key is not working to close the dialog when I use Safari and VoiceOver.

**Comment by terracoda** [comment]
I found the dynamic Scene Summary text while testing the sim with Safari and VoiceOver. Awesome!

**Comment by jessegreenberg** [comment]
Escape key closes the help dialog again.  Since that is fixed and the jump keys were fixed, closing this issue.

**Comment by terracoda** [comment]
@jesse The last section in the Hot Keys dialog does not match the original content (above). The heading  "Hot keys for jumping" is accidentally repeated. It should have been "Grab Balloon." Instead of fixing, we can actually change it because both the Enter Key and Space bar work to grab and release the Balloon. I think it is better to combine the Release section with the Grab section.

H2: Grab and Release Balloon
- Press Space bar or Enter on the Grab Balloon button to pick up Balloon.
- When balloon is grabbed press Space bar or Enter to release, or let go.
- Pressing the Tab key also releases the Balloon as you go to next item.

@jessegreenberg, also, note that the directions on how to close the dialog are missing. Most users seem to do this intuitively, but it might be ok to be explicit.

_Press Escape or Close button to exit dialog and return to sim._

**Comment by jessegreenberg** [comment]
Good catch @terracoda, thanks!  I will combine the Release section with the grab section to fix the issue.

For the direction on how to close the dialog, where should this exist in the dialog?  Near the top by the h1 or more as an on-demand description for the 'Close' button?

**Comment by terracoda** [comment]
@jessegreenberg, the direction to close the dialog can be at the very end. Most users know how to do this intuitively, so it is not something that has to take up prime reading space, and is only there in case a user does not know, or reaches the end is not sure what to do.

**Comment by jessegreenberg** [comment]
This issue is stale now, we can close.  More work on keyboard help information is being continued in other issues.
